### PR TITLE
Leave Application - validate optional leaves

### PR DESCRIPTION
**Pull request by rkurungadam**
https://github.com/frappe/erpnext/issues/13614

### CODE REVIEW

**Review by nabinhait — CHANGES_REQUESTED** [review]
- [ ] We have added a sudo test `test_optional_leave` in test_leave_application.py file. Please make the necessary changes to pass it.
- [ ] Add docs

File: erpnext/hr/doctype/leave_application/leave_application.py [hunk 1]
```diff
@@ -373,6 +383,10 @@ def is_lwp(leave_type):
 	lwp = frappe.db.sql("select is_lwp from `tabLeave Type` where name = %s", leave_type)
 	return lwp and cint(lwp[0][0]) or 0
 
+def is_optional_leave(leave_type):
+	optional_leave = frappe.db.sql("select is_optional_leave from `tabLeave Type` where name = %s", leave_type)
```
[the remark this answers is not in the document]
Comment (rkurungadam) [reply erpnext/hr/doctype/leave_application/leave_application.py]: found a smilar method is_lwp(), so thought db.sql is more optimal, change this too?

File: erpnext/hr/doctype/leave_application/leave_application.py [hunk 2]
```diff
@@ -195,6 +197,14 @@ def validate_attendance(self):
 			frappe.throw(_("Attendance for employee {0} is already marked for this day").format(self.employee),
 				AttendanceAlreadyMarkedError)
 
+	def validate_optional_leave(self):
+		day = getdate(self.from_date)
+		holiday_list = frappe.get_value("Leave Type", self.leave_type, "holiday_list")
+		while day <= getdate(self.to_date):
```
[the remark this answers is not in the document]
Comment (rkurungadam) [reply erpnext/hr/doctype/leave_application/leave_application.py]: do we need holiday list in leave type? by this design leave types will have to be updated on period change. Holidays are fetched by get_holiday_list_for_employee in other places, let me know.

**Comment by nabinhait** [comment]
Can you please rebase and fix conflicts?

**Pull request closed without merging**